frontend/server/src/main/java/com/amazonaws/ml/mms/http/HttpRequestHandler.java

```diff
@@ -75,6 +88,77 @@ protected void channelRead0(ChannelHandlerContext ctx, FullHttpRequest req) {
+            QueryStringDecoder decoder,
+            String method)
+            throws IOException {
+        switch (method) {
+            case "GET":
+                endpoint.doGet(
+                        new ModelServerRequest(req, decoder),
+                        new ModelServerResponse(rsp),
+                        new ModelServerContext());
+                break;
+            case "PUT":
+                endpoint.doPut(
+                        new ModelServerRequest(req, decoder),
+                        new ModelServerResponse(rsp),
+                        new ModelServerContext());
+                break;
+            case "DELETE":
+                endpoint.doDelete(
+                        new ModelServerRequest(req, decoder),
+                        new ModelServerResponse(rsp),
+                        new ModelServerContext());
+                break;
+            case "POST":
+                endpoint.doPost(
+                        new ModelServerRequest(req, decoder),
+                        new ModelServerResponse(rsp),
+                        new ModelServerContext());
+                break;
+            default:
+                throw new ServiceUnavailableException("Invalid HTTP method received");
+        }
+    }
+
+    protected void handleCustomEndpoint(
+            ChannelHandlerContext ctx,
+            FullHttpRequest req,
+            String[] segments,
+            QueryStringDecoder decoder) {
+        ModelServerEndpoint endpoint = endpointMap.get(segments[1]);
+        Runnable r =
+                () -> {
+                    Long start = System.currentTimeMillis();
+                    FullHttpResponse rsp =
+                            new DefaultFullHttpResponse(
+                                    HttpVersion.HTTP_1_1, HttpResponseStatus.OK, false);
+                    try {
+                        run(endpoint, req, rsp, decoder, req.method().toString());
+                        NettyUtils.sendHttpResponse(ctx, rsp, true);
+                        logger.info(
+                                "Running \"{}\" endpoint took {} ms",
+                                segments[0],
+                                System.currentTimeMillis() - start);
+                    } catch (IOException ioe) {
+                        NettyUtils.sendError(ctx, HttpResponseStatus.INTERNAL_SERVER_ERROR, ioe);
+                        ByteArrayOutputStream ps = new ByteArrayOutputStream();
+                        ioe.printStackTrace(new PrintStream(ps));
+                        logger.error("I/O error while running the custom endpoint.", ioe);
+                    } catch (Throwable e) {
+                        NettyUtils.sendError(ctx, HttpResponseStatus.NOT_IMPLEMENTED, e);
+                        ByteArrayOutputStream ps = new ByteArrayOutputStream();
+                        e.printStackTrace(new PrintStream(ps));
+                        logger.error("Unknown exception", e);
```

Comment:
ps is not used anywhere, we can remove the code.

Reply:
Done